akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -237,17 +268,7 @@ private[projection] class R2dbcOffsetStore(
   private val inflight = new AtomicReference(Map.empty[Pid, SeqNr])
 
   // To avoid delete requests when no new offsets have been stored since previous delete
-  private val idle = new AtomicBoolean(false)
-
-  // To trigger next deletion after in-memory eviction
-  private val triggerDeletion = new AtomicBoolean(false)
-
-  if (!settings.deleteInterval.isZero && !settings.deleteInterval.isNegative)
-    system.scheduler.scheduleWithFixedDelay(
-      settings.deleteInterval,
-      settings.deleteInterval,
-      () => deleteOldTimestampOffsets(),
-      system.executionContext)
+  private val triggerDeletionPerSlice = new ConcurrentHashMap[Int, java.lang.Boolean]
```

Comment:
Needs to use `java.lang.Boolean`? Should be able to just use Scala Boolean?

Reply:
I was thinking that it would be boxing the Scala Boolean, and this is touched for each save offset. Is such boolean boxing is optimized?

Reply:
The boxing should use `java.lang.Boolean.valueOf`, which uses the constants. But didn't register the presence check with null, where I think scala boolean will always be unboxed automatically as well.

Reply:
I guess we could also create a concurrent bitset, based on something like AtomicLongArray, since it's always slice ranges. And for the delete when initially undefined, start with all bits flipped. That would probably be the most optimised for this.